src/db.js

```diff
@@ -9,6 +9,7 @@ import { timestamp } from './util'
 
 /** Returns true if the app is running as a test. */
 const isTest = () => process.env.NODE_ENV === 'test'
+
 if (isTest()) {
   Dexie.dependencies.indexedDB = require('fake-indexeddb')
   Dexie.dependencies.IDBKeyRange = require('fake-indexeddb/lib/FDBKeyRange')
```

Comment:
Is it possible to move these into the test files? We should try to reduce test detection in the regular codebase if possible. I had put them here only because I could not get them working in the tests.

Reply:
I was thinking the same.  Instead of directly creating `dexie` object inside `db`. Let's create a class and abstract all the db functions inside. Then we can inject `dexie` from the constructor. So we can easily pass `dexie` object with fake index db for the test. What do you think ?

Reply:
I like the idea of not automatically creating the `dexie` object in `db.js`. I prefer not to use classes, but presumably we could do the same thing with a simple factory function.

Are you imagining `db.js` would still contain a singleton instance? It seems like this would be still be required wherever `db.js` is imported, since the db functions operate on the `db` instance.